Lib/urllib/request.py

```diff
@@ -163,18 +163,10 @@ def urlopen(url, data=None, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
 
     The *cadefault* parameter is ignored.
 
-    This function always returns an object which can work as a context
-    manager and has methods such as
 
-    * geturl() - return the URL of the resource retrieved, commonly used to
-      determine if a redirect was followed
-
-    * info() - return the meta-information of the page, such as headers, in the
-      form of an email.message_from_string() instance (see Quick Reference to
-      HTTP Headers)
-
-    * getcode() - return the HTTP status code of the response.  Raises URLError
-      on errors.
+    This function always returns an object which can work as a
+    :term:`context manager` and has the properties `url`, `headers`, and `status`.
+    See :class:`urllib.response.addinfourl` for more detail on these properties.
```

Comment:
No reST in docstrings please.

Reply:
Fixed, thanks. What is the relationship between docstrings and documentation in python? Are they supposed to be kept in sync?

Reply:
Python docs don’t use sphinx autodoc to extract docstrings, and the texts are also not usually a copy-paste.

Docstrings should give enough info about how to use the function in an editor tooltip or when calling `help(...)` or `pydoc`.

ReST documentations can contain into, detailed info, comparisons, links, etc.